Moonshot/player/player.gd

```diff
@@ -174,7 +174,7 @@ func damagetile_check() -> void:
 
 func take_damage(damage, attack_dir, is_damage_tile: bool = false) -> void:
 	if not invulnerable:
-		Utils.player_stats.health -= damage
+		Utils.player_stats.take_damage(damage)
```

Comment:
Related to the other coments, This line should just remove HP, and then a trigger to modify firerate should happen after line 182. It would also mean we could specify whether damage tiles modify the firerate stat differently.